r332965 | emaste | 2018-04-24 21:26:58 +0200 (Tue, 24 Apr 2018) | 8 lines

lldb: remove assertion that target_arch is FreeBSD

The target is not necessarily a FreeBSD binary - for example, it may be
a Linux binary running under the linuxulator.  Basic ptrace (live)
debugging already worked in this case, except for the assertion.

Sponsored by:	Turing Robotic Industries Inc.

Index: tools/lldb/source/Plugins/Process/FreeBSD/FreeBSDThread.cpp
===================================================================
--- lldb/source/Plugins/Process/FreeBSD/FreeBSDThread.cpp	(revision 332964)
+++ lldb/source/Plugins/Process/FreeBSD/FreeBSDThread.cpp	(revision 332965)
@@ -169,7 +169,6 @@ lldb::RegisterContextSP FreeBSDThread::GetRegister
     RegisterInfoInterface *reg_interface = nullptr;
     const ArchSpec &target_arch = GetProcess()->GetTarget().GetArchitecture();
 
-    assert(target_arch.GetTriple().getOS() == llvm::Triple::FreeBSD);
     switch (target_arch.GetMachine()) {
     case llvm::Triple::aarch64:
       reg_interface = new RegisterInfoPOSIX_arm64(target_arch);
